source/features/restore-file.tsx

```diff
@@ -79,32 +74,47 @@ async function handleRestoreFileClick(event: delegate.Event<MouseEvent, HTMLButt
 
 	filesRestored.add(menuItem);
 
-	menuItem.textContent = 'Restoring…';
-	event.preventDefault();
-	event.stopPropagation();
-
-	try {
-		const filePath = menuItem.closest<HTMLDivElement>('[data-path]')!.dataset.path!;
+	async function restoreFile(filePath: string, progress: (message: string) => void): Promise<void> {
+		progress('Restoring...');
 		const file = await getFile(filePath);
 
 		if (!file) {
 			// The file was created by this PR.
 			// This code won’t be reached if `highlight-deleted-and-added-files-in-diffs` works.
-			showError(menuItem, 'Nothing to restore. Delete file instead');
-			return;
+			throw new Error('Nothing to restore. Delete file instead');
 		}
 
 		if (file.isTruncated) {
-			showError(menuItem, 'Restore failed: File too big');
-			return;
+			throw new Error('Restore failed: File too big');
 		}
 
-		await commitFileContent(menuItem, file.text, filePath);
+		await commitFileContent(menuItem, file.text, filePath, progress);
+	}
 
+	try {
+		const filePath = menuItem.closest<HTMLDivElement>('[data-path]')!.dataset.path!;
+		// Show toast while restoring
+		const progress = async (message: string): Promise<void> => {
+			void showToast(async () => task, {
+				message,
+				doneMessage: 'Restored!',
+				onDone: ToastOnDoneState.success,
+			});
+		};
+
+		const task = restoreFile(filePath, progress);
 		// Hide file from view
+		await task;
 		menuItem.closest('.file')!.remove();
 	} catch (error: unknown) {
-		showError(menuItem, 'Restore failed. See console for details');
+		menuItem.disabled = true;
+		menuItem.style.background = 'none'; // Disables hover background color
+		void showToast(async () => {
+			// Fake a task to show the error message
+		}, {
+			doneMessage: 'Restore failed. See console for details',
+			onDone: ToastOnDoneState.error,
+		});
```

Comment:
This is probably automatically handled by the toast. There should not be any empty task functions

Reply:
As far as I know, It is not automatically handled by the toast.

Reply:
You're right, it's missing a catch after the `return await task()`